tests/frontend/test_parameters.py

```diff
@@ -206,3 +225,127 @@ def test_parameter_unary_negation(self, p):
+            gate(r, phi) | q[0]
+
+        # create numeric values
+        mapping = {'r': tf.Variable(0.1), 'phi': tf.Variable(0.1)}
+        prog.bind_params(mapping)
+
+        # verify bound parameters are correct
+        assert prog.free_params['r'].val is mapping['r']
+        assert prog.free_params['phi'].val is mapping['phi']
+
+        # assert executed program is constructed correctly
+        global applied
+        cmds = []
+        applied = []
+
+        with monkeypatch.context() as m:
+            m.setattr(sf.LocalEngine, "_run_program", mock_run_prog)
+            eng = sf.LocalEngine(backend)
+            result = eng.run(prog, args=mapping)
+
+        assert isinstance(applied[0].op, gate)
+        assert applied[0].op.p[0].val == mapping["r"]
+        assert applied[0].op.p[1].val == mapping["phi"]
+
+    @pytest.mark.parametrize("gate", [sf.ops.BSgate, sf.ops.S2gate])
+    def test_two_mode_gate_complex_phase(self, backend, gate, monkeypatch):
+        """Test two mode gates with complex phase arguments"""
+        import tensorflow as tf
+        prog = sf.Program(2)
+
+        # create symbolic parameters
+        r = prog.params('r')
+        phi = prog.params('phi')
+
+        with prog as q:
+            gate(r, phi) | q
+
+        # create numeric values
+        mapping = {'r': tf.Variable(0.1), 'phi': tf.Variable(0.1)}
+        prog.bind_params(mapping)
+
+        # verify bound parameters are correct
+        assert prog.free_params['r'].val is mapping['r']
+        assert prog.free_params['phi'].val is mapping['phi']
+
+        # assert executed program is constructed correctly
+        global applied
+        cmds = []
+        applied = []
+
+        with monkeypatch.context() as m:
+            m.setattr(sf.LocalEngine, "_run_program", mock_run_prog)
+            eng = sf.LocalEngine(backend)
+            result = eng.run(prog, args=mapping)
+
+        assert isinstance(applied[0].op, gate)
+        assert applied[0].op.p[0].val == mapping["r"]
+        assert applied[0].op.p[1].val == mapping["phi"]
+
+    def test_zgate(self, backend, hbar, monkeypatch):
+        """Test the momentum displacement gate"""
+        import tensorflow as tf
+        prog = sf.Program(1)
+
+        # create symbolic parameters
+        p = prog.params('p')
+
+        with prog as q:
+            sf.ops.Zgate(p) | q[0]
+
+        # create numeric values
+        mapping = {'p': tf.Variable(0.1)}
+        prog.bind_params(mapping)
+
+        # verify bound parameters are correct
+        assert prog.free_params['p'].val is mapping['p']
+
+        # assert executed program is constructed correctly
+        global applied
+        cmds = []
+        applied = []
+
+        with monkeypatch.context() as m:
+            m.setattr(sf.LocalEngine, "_run_program", mock_run_prog)
+            eng = sf.LocalEngine(backend)
+            result = eng.run(prog, args=mapping)
+
+        assert isinstance(applied[0].op, sf.ops.Dgate)
```

Comment:
So this would be an example for decomposition, correct? Worth adding this to the docstring

Reply:
Have updated the docstring and test name!